### PR TITLE
New ruby hash

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

**Comment by mvidner** [comment]
It could also be done by tweaking the original regex, but I will resist the urge :)
LGTM.

File: ytools/y2tool/find-unused-published [hunk 1]
```diff
@@ -14,9 +14,17 @@ usage() {
 published() {
     # lines like
     #    publish :function => :Add, :type => "boolean ()"
+    # or new ruby hash style
+    #    publish function: :Add, type: "boolean ()"
     # transformed to
     #    Add
-    sed -n -e '/publish/s/[^>]*> :\([^,]*\),.*/\1/;T;p' "$1"
+    RES=`sed -n -e '/publish/s/[^>]*> :\([^,]*\),.*/\1/;T;p' "$1"`
+    # file use new ruby format
```
Comment (lslezak) [inline ytools/y2tool/find-unused-published]: NP: typo: `file uses`

**Comment by lslezak** [comment]
@mvidner later we could add this check to `rubocop-yast`...

**Comment by jreidinger** [comment]
thanks. BTW we should look closer on performance, it take quite a lot of time.

**Comment by mvidner** [comment]
@jreidinger Re performance: yes, it is bad, because for each published symbol it runs a separate recursive grep. That should be easily fixable by constructing a `(regex|with|alternatives)` for a single grep.

**Pull request merged**